module-apps/application-phonebook/ApplicationPhonebook.cpp

```diff
@@ -149,19 +145,15 @@ namespace app
 
     void ApplicationPhonebook::onSearchRequest(const std::string &searchFilter)
     {
-        auto searchModel = std::make_unique<PhonebookModel>(this, searchFilter);
-
-        LOG_DEBUG("Search results count: %d", searchModel->requestRecordsCount());
-        if (searchModel->requestRecordsCount() > 0) {
-            auto main_window =
-                dynamic_cast<gui::PhonebookMainWindow *>(windowsStack.get(gui::name::window::main_window));
-            if (main_window == nullptr) {
-                LOG_ERROR("Failed to get main window.");
-                return;
-            }
+        auto phonebookModel = std::make_unique<PhonebookModel>(this, searchFilter);
 
-            if (main_window->isSearchRequested()) {
-                searchModel->messagesSelectCallback = [=](gui::PhonebookItem *item) {
+        LOG_DEBUG("Search results count: %d", phonebookModel->requestRecordsCount());
+        if (not searchModel) {
+            LOG_ERROR("application not build, model not available");
```

Comment:
so that is only print ? It can happen and we do not handle it ? maybe assert or something?

Reply:
assert addedz